c++/src/Compression.cc

```diff
@@ -633,9 +642,10 @@ DIAGNOSTIC_PUSH
   ZlibDecompressionStream::ZlibDecompressionStream
                    (std::unique_ptr<SeekableInputStream> inStream,
                     size_t bufferSize,
-                    MemoryPool& _pool
+                    MemoryPool& _pool,
+                    ReaderMetrics& metrics
```

Comment:
Rename the argument to _metrics, this is why the MasOS build fails. Same for constructors below.

> [ 48%] Building CXX object c++/src/CMakeFiles/orc.dir/Compression.cc.o
/Users/runner/work/orc/orc/c++/src/Compression.cc:646:36: error: parameter 'metrics' shadows member inherited from type 'DecompressionStream' [-Werror,-Wshadow-field]
                    ReaderMetrics& metrics
                                   ^
/Users/runner/work/orc/orc/c++/src/Compression.cc:396:20: note: declared here
    ReaderMetrics& metrics;
                   ^
/Users/runner/work/orc/orc/c++/src/Compression.cc:767:36: error: parameter 'metrics' shadows member inherited from type 'DecompressionStream' [-Werror,-Wshadow-field]
                    ReaderMetrics& metrics
                                   ^
/Users/runner/work/orc/orc/c++/src/Compression.cc:396:20: note: declared here
    ReaderMetrics& metrics;
                   ^
/Users/runner/work/orc/orc/c++/src/Compression.cc:816:46: error: parameter 'metrics' shadows member inherited from type 'DecompressionStream' [-Werror,-Wshadow-field]
                              ReaderMetrics& metrics
                                             ^
/Users/runner/work/orc/orc/c++/src/Compression.cc:396:20: note: declared here
    ReaderMetrics& metrics;
                   ^
/Users/runner/work/orc/orc/c++/src/Compression.cc:861:43: error: parameter 'metrics' shadows member inherited from type 'DecompressionStream' [-Werror,-Wshadow-field]
                           ReaderMetrics& metrics
                                          ^
/Users/runner/work/orc/orc/c++/src/Compression.cc:396:20: note: declared here
    ReaderMetrics& metrics;
                   ^
/Users/runner/work/orc/orc/c++/src/Compression.cc:895:43: error: parameter 'metrics' shadows member inherited from type 'DecompressionStream' [-Werror,-Wshadow-field]
                           ReaderMetrics& metrics
                                          ^
/Users/runner/work/orc/orc/c++/src/Compression.cc:396:20: note: declared here
    ReaderMetrics& metrics;
                   ^
/Users/runner/work/orc/orc/c++/src/Compression.cc:[1201](https://github.com/apache/orc/runs/6884140918?check_suite_focus=true#step:5:1202):44: error: parameter 'metrics' shadows member inherited from type 'DecompressionStream' [-Werror,-Wshadow-field]
                            ReaderMetrics& metrics)
                                           ^
/Users/runner/work/orc/orc/c++/src/Compression.cc:396:20: note: declared here
    ReaderMetrics& metrics;
                   ^
6 errors generated.
make[2]: *** [c++/src/CMakeFiles/orc.dir/Compression.cc.o] Error 1
make[1]: *** [c++/src/CMakeFiles/orc.dir/all] Error 2
make: *** [all] Error 2